caom2-search-server/src/main/resources/META-INF/resources/js/cadc.search.app.js

```diff
@@ -1066,62 +1066,152 @@
+                  $input.prop('type', 'hidden')
+                  $input.prop('name', 'uri')
+                  $input.prop('id', $nextPlaneURI)
+                  $input.val($nextPlaneURI)
+
+                  downloadForm.append($input)
+                }
               }
 
               // Story 1566, when all 'Product Types'
               // checkboxes are checked, do not send any
+              // both (2739) - although I wonder if the 'all checked' check
+              //being done twice is necessary (after the submit, for example.)
               var allChecked =
-                      downloadForm
-                          .find('input.product_type_option_flag')
-                          .not(':checked').length === 0
+                    downloadForm
+                      .find('input.product_type_option_flag')
+                      .not(':checked').length === 0
               if (allChecked) {
                 // disable all 'Product Types' checkboxes
                 $.each(
-                    downloadForm.find('input.product_type_option_flag:checked'),
-                    function () {
-                      $(this).prop('disabled', true)
-                    }
+                  downloadForm.find('input.product_type_option_flag:checked'),
+                  function () {
+                    $(this).prop('disabled', true)
+                  }
                 )
               }
 
-              window.open('', 'DOWNLOAD', '')
-              downloadForm.submit()
+              // Now get down to submitting the data
+              if (fromInputFile) {
+                // iterate through downloadTuples and make the badgerfish json
+                var badgerfishTuples = "";
+                for (i=0; i<downloadTuples.length; i++) {
+                  var tupleJSON = "{\"tuple\":{\"tupleID\":{\"$\":\"" + downloadTuples[i].tupleID + "\"}," +
+                    "\"shape\":{\"$\":\"" + downloadTuples[i].shape + "\"}," +
+                    "\"label\":{\"$\":\"" + downloadTuples[i].label + "\"}}}";
+                  badgerfishTuples += tupleJSON + ","; // add another badgerfish beast on here..
+                }
+                // trim off last ','
+                badgerfishTuples = badgerfishTuples.substr(0, badgerfishTuples.length-1);
+
+                // create payload item
+                var jsonTuples = "{\"tupleList\":{\"$\":[" + badgerfishTuples + "]}}";
+
+                var multiPartData = new FormData()
+                var runID = downloadForm.find("input[name='runid']").val()
+                if (runID !=  null) {
+                  multiPartData.append('runid', runID)
+                }
+
+                // 'Blob' type is requred to have the 'filename="blob" parameter added
+                // to the multipart section, and have the Content-type header added
+                // so that the web service can correctly parse JSON payload sent
+                multiPartData.append('blob', new Blob([jsonTuples], {
```

Comment:
Then here just issue a `JSON.stringify` call to the JSON object.

Reply:
done